cmd/tailscale/cli/file.go

```diff
@@ -203,7 +203,26 @@ func discoverPeerAPIBase(ctx context.Context, ipStr string) (base string, lastSe
 			return ft.PeerAPIURL, lastSeen, isOffline, nil
 		}
 	}
-	return "", time.Time{}, false, errors.New("target seems to be running an old Tailscale version")
+
+	// Try a bit harder for a nicer error message.
+	found := false
+	if st, err := tailscale.Status(ctx); err == nil && st.Self != nil {
+		for _, peer := range st.Peer {
+			for _, pip := range peer.TailscaleIPs {
+				if pip == ip {
+					found = true
+					if peer.UserID != st.Self.UserID {
+						return "", time.Time{}, false, fmt.Errorf("%s (%s) is owned by a different user; can only send files to your own devices",
+							ipStr, peer.DNSName)
+					}
+				}
+			}
+		}
+	}
+	if found {
+		return "", time.Time{}, false, errors.New("target seems to be running an old Tailscale version")
```

Comment:
Can we substitute a DNSName or other human-readable name for the word "target"?

Reply:
Addressed by making caller always include the target name as typed by the user.

It'll still as target, though, but it'll be in context. Also, it matches the --targets flag.